### PR TITLE
Add more verification item

### PR DESCRIPTION
**Pull request by rosona**
- [x] Block/Header check

### CODE REVIEW

File: src/AElf.Contracts.TestKit/TransactionExecutor.cs [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 using AElf.Kernel.SmartContractExecution.Application;
-using AElf.Kernel.TransactionPool.Infrastructure;
 using Google.Protobuf;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace AElf.Contracts.TestKit
 {
     public class TransactionExecutor : ITransactionExecutor
     {
+        public ECKeyPair KeyPair { get; set; } = CryptoHelpers.GenerateKeyPair();
```
Comment (loning) [inline src/AElf.Contracts.TestKit/TransactionExecutor.cs]: should not put sign logic in executing

File: src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs [hunk 2]
```diff
@@ -80,28 +81,42 @@ public ValidateNextTimeBlockValidationException(string message, Exception inner)
...
 
             if (block.Body.TransactionsCount == 0)
-            {
                 return false;
-            }
+
+            if (_blockchainServce.GetChainId() != block.Header.ChainId)
+                return false;
+
+            if (block.Height != 1 && !block.VerifySignature())
```
Comment (loning) [inline src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs]: first block should also verify signature

File: src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs [hunk 3]
```diff
@@ -80,28 +81,42 @@ public ValidateNextTimeBlockValidationException(string message, Exception inner)
...
 
             if (block.Body.CalculateMerkleTreeRoots() != block.Header.MerkleTreeRootOfTransactions)
-            {
                 return false;
-            }
 
-            // TODO: Time span maybe configurable.
-            if (block.Header.Time.ToDateTime() - DateTime.UtcNow > TimeSpan.FromMilliseconds(2000))
-            {
+            if (block.Height != 1 && block.Header.Time.ToDateTime() - DateTime.UtcNow > KernelConsts.AllowedFutureBlockTimeSpan)
```
Comment (loning) [inline src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs]: if we have const GenesisBlockHeight, we should use it

File: src/AElf.Contracts.TestKit/TransactionExecutor.cs (hunk 1 above)
Comment (rosona) [reply src/AElf.Contracts.TestKit/TransactionExecutor.cs]: Add account class for block sign

File: src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs (hunk 3 above)
Comment (rosona) [reply src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs]: Fixed

File: src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs (hunk 2 above)
Comment (rosona) [reply src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs]: The first block will not be broadcast to others. In order to ensure the first block of each BP is the same, no signature is added.

File: src/AElf.Contracts.TestKit/Account.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace AElf.Contracts.TestKit
+{
+    public interface IAccount
+    {
+        byte[] Sign(byte[] data);
+        byte[] GetPublicKey();
+    }
+
+    public class Account : IAccount
```
Comment (loning) [inline src/AElf.Contracts.TestKit/Account.cs]: directly use IAccountService

File: src/AElf.Contracts.TestKit/TransactionExecutor.cs [hunk 5]
```diff
@@ -27,10 +26,13 @@ public async Task ExecuteAsync(Transaction transaction)
             var preBlock = await blockchainService.GetBestChainLastBlockHeaderAsync();
             var minerService = _serviceProvider.GetRequiredService<IMiningService>();
             var blockAttachService = _serviceProvider.GetRequiredService<IBlockAttachService>();
+            var account = _serviceProvider.GetRequiredService<IAccount>();
```
Comment (loning) [inline src/AElf.Contracts.TestKit/TransactionExecutor.cs]: Pass a transaction signed 
or pass method name and params, and create a new transaction in this function. and rename the class name to GenerateBlockAndExecuteTransactions

Comment (rosona) [reply src/AElf.Contracts.TestKit/TransactionExecutor.cs]: Solved by offline communication: For this keep old method name

**Pull request merged**